ls1intum/Artemis
# `Lectures`: Add new lecture unit for online resources

### Checklist
#### General


#### Server
#### Client

### Motivation and Context



Instructors want to link to external online content in their lectures. Currently, they have to use a text unit (Markdown editor) as a workaround for that.

### Description


We add a new lecture unit type for online resources. It consists of a title, description, link, and release date.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis as Instructor
2. Navigate to Course Administration
3. Create a lecture and add lecture units of type `Online` (pasting a URL in the link field should autofill the rest)
4. Go to Course Overview
5. Checkout the lecture and its lecture units

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| create-online-unit.component.ts | 69% | 93% |
| edit-online-unit.component.ts | 75% | 97% |
| online-unit-form.component.ts | 80% | 100% |
| online-unit.component.ts | 80% | 92% |
| onlineUnit.service.ts | 75% | 100% |

### Screenshots



<img src="https://user-images.githubusercontent.com/102237363/170886867-4f955b87-1a8a-4c0d-b050-cf0d24ea8469.png" width="800" />

<img src="https://user-images.githubusercontent.com/102237363/170886914-34b0db07-4e6e-40e0-8c67-4fa6af0f8357.png" width="800" />

<img src="https://user-images.githubusercontent.com/102237363/170886896-bbd33729-e631-4468-98ea-d614a5007de1.png" width="800" />
